File: client/src/app/modules/quotation/services/debour.del.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { DebourDel } from '../model/DebourDel';
import { Provision } from '../model/Provision';

@Injectable({
  providedIn: 'root'
})
export class DebourDelService extends AppRestService<DebourDel>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getDebourByProvision(provision: Provision) {
    return this.getList(new HttpParams().set("idProvision", provision.id), "debour");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/DebourDelService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.DebourDel;
import com.jss.osiris.modules.osiris.quotation.model.Provision;

public interface DebourDelService {
    public List<DebourDel> getDebourByProvision(Provision provision);
}
